app/controllers/api/v3/timelines/channels_controller.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+class Api::V3::Timelines::ChannelsController < Api::BaseController
+  before_action :set_channel
+
+  after_action :insert_pagination_headers, only: [:show], unless: -> { @statuses.empty? }
+
+  rescue_from Mammoth::Channels::NotFound do |e|
+    render json: { error: e.to_s }, status: 404
+  end
+
+  def show
+    @statuses = cached_channel_statuses
+    render json: @statuses,
+           each_serializer: REST::StatusSerializer,
+           relationships: StatusRelationshipsPresenter.new(@statuses, current_user&.account_id)
+  end
+
+  private
+
+  def set_channel
+    @mammoth = Mammoth::Channels.new
+    @channel = @mammoth.find(channel_id_param)
+  end
+
+  def cached_channel_statuses
+    cache_collection channel_statuses, Status
+  end
+
+  def channel_statuses
+    channel_feed.get(DEFAULT_STATUSES_LIST_LIMIT,
```

Comment:
Not sure where DEFAULT_STATUSES_LIST_LIMIT is defined, but maybe worth adding a note that it's used for both lists and channels (?). Not sure.

Reply:
that's a good question. Rails has this idea of a 'base controller' for methods and variables that are used across multiple controllers that controllers inherit from `class Api::V3::Timelines::ChannelsController < Api::BaseController` I added a comment to both sides noting that it's also used for channels too.